server/src/main/java/org/elasticsearch/common/util/Maps.java

```diff
@@ -199,13 +199,13 @@
      * unmodifiable sorted map. The resulting read-only view through the unmodifiable sorted map is a sorted map.
      *
      * @param <T> the type of the input elements
-     * @return an unmodifiable map where the underlying map is sorted
+     * @return an unmodifiable {@link NavigableMap} where the underlying map is sorted
      */
-    public static <T, K, V> Collector<T, ?, SortedMap<K, V>> toUnmodifiableSortedMap(Function<T, ? extends K> keyMapper,
-                                                                                     Function<T, ? extends V> valueMapper) {
+    public static <T, K, V> Collector<T, ?, NavigableMap<K, V>> toUnmodifiableSortedMap(Function<T, ? extends K> keyMapper,
```

Comment:
Let me know whether it makes sense to rename the method to `toUnmodifiableSortedMap`

Reply:
Yes, I think it makes sense